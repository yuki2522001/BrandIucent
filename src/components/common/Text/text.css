.text--blur {
  font-family: "Poppins";
  text-transform: uppercase;
  color: var(--bg-third);
  font-weight: 600;
  font-size: 24px;
  line-height: 36px;
}

.text--large,
.text--dark {
  font-weight: 600;
  font-size: 36px;
  line-height: 54px;
}

.text--large {
  color: var(--text-primary);
}

.text--dark {
  color: var(--dark);
}

.text--medium {
  font-weight: 600;
  font-size: 36px;
  line-height: 54px;
  color: var(--text-primary);
  width: 561px;
}

.text--normal,
.text--highlight {
  font-family: "Poppins";
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
}

.text--normal {
  color: var(--dark);
}

.text--highlight {
  color: var(--text-primary);
}