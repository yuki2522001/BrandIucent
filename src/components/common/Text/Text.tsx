import React, { memo } from "react";
import "./text.css";

interface TitleProps {
  size?: string;
  decoration?: string;
  text?: string;
}

const Text: React.FC<TitleProps> = memo(({ text, size, decoration }) => {
  let className = "";
  switch (size) {
    case "normal":
      className = "text--normal";
      break;
    case "medium":
      className = "text--medium";
      break;
    case "large":
      className = "text--large";
      break;
    default:
      break;
  }

  switch (decoration) {
    case "blur":
      className = "text--blur";
      break;
    case "highlight":
      className = "text--highlight";
      break;
    case "dark":
      className = "text--dark";
      break;
    default:
      break;
  }

  return (
    <p data-testid="text" className={className}>
      {text}
    </p>
  );
});

export default Text;
