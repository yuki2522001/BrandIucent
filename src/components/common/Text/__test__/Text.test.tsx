import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Text from "../Text";

describe("component Text", () => {
  test("should render component Text size 'blur' ", () => {
    const { getByTestId } = render(<Text size="blur" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'normal' ", () => {
    const { getByTestId } = render(<Text size="normal" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'color' ", () => {
    const { getByTestId } = render(<Text size="color" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'normal' ", () => {
    const { getByTestId } = render(<Text size="blur" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'medium' ", () => {
    const { getByTestId } = render(<Text size="medium" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'large' ", () => {
    const { getByTestId } = render(<Text size="large" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("should render component Text size 'large-dark' ", () => {
    const { getByTestId } = render(<Text size="large-dark" text="Offers" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Text text="Offers" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
