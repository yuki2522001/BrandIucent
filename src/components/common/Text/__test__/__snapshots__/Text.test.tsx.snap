// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Text matches snapshot 1`] = `
<DocumentFragment>
  <p
    class=""
    data-testid="text"
  >
    Offers
  </p>
</DocumentFragment>
`;
